File: number3/number3.cpp
#include <iostream>
#include <vector>
bool input(int& amount, std::vector<int>& numbers);
void palindrome(int amount, std::vector<int> numbers);
int main(){
	int amount=0;
	std::vector<int> numbers;
	if(input(amount, numbers)){
		palindrome(amount, numbers);
	}
	return 0;
}
bool input(int & amount, std::vector<int>& numbers){
	int isDone=1;
	if(std::cin>>amount && amount>-1){
		for(int i=0;i<amount;++i){
			int x;
			if(std::cin>>x && x>0){
				numbers.push_back(x);
			}
			else{
				isDone=0;
			}
		}
	}
	else{
		isDone=0;
	}
	if(isDone==0){
		std::cout<<"Введите коректное количество чисел, числа должны быть положительные"<<std::endl;
	}
	return isDone;
}
void palindrome(int amount, std::vector<int> numbers){
	int number=0, unit=0, tens=0, hung=0;
	for(int i=0; i<amount;++i){
		if(numbers[i]>99 && numbers[i]<1000){
			unit=numbers[i]%10;
			tens=numbers[i]%100-unit;
			hung=numbers[i]%1000-tens-unit;
			number=(unit*100)+(tens)+(hung/100);
			if(number==numbers[i]){
				std::cout<<"Yes ";
			}
			else{
				std::cout<<"No ";
			}
		}
		else{
			std::cout<<"- ";
		}
	}
	std::cout<<std::endl;
}

File: number3/number3.go
package main

import "fmt"

func input(amount *int, numbers *[]int) int {
	isDone := 1
	var n int
	if _, err := fmt.Scan(&n); err == nil && n > -1 {
		*amount = n
		for i := 0; i < n; i++ {
			var x int
			if _, err := fmt.Scan(&x); err == nil && x > 0 {
				*numbers = append(*numbers, x)
			} else {
				isDone = 0
			}
		}
	} else {
		isDone = 0
	}
	if isDone == 0 {
		fmt.Println("Введите коректное количество чисел, числа должны быть положительные")
	}
	return isDone
}

func palindrome(amount int, numbers []int) {
	var number, unit, tens, hung int
	for i := 0; i < amount; i++ {
		if numbers[i] > 99 && numbers[i] < 1000 {
			unit = numbers[i] % 10
			tens = numbers[i]%100 - unit
			hung = numbers[i]%1000 - tens - unit
			number = (unit * 100) + tens + (hung / 100)
			if number == numbers[i] {
				fmt.Print("Yes ")
			} else {
				fmt.Print("No ")
			}
		} else {
			fmt.Print("- ")
		}
	}
	fmt.Println()
}

func main() {
	amount := 0
	numbers := []int{}
	if input(&amount, &numbers) == 1 {
		palindrome(amount, numbers)
	}
}
